File: Logic/src/models/Invoice.java
package models;

import serviceproviders.PaymentState;

public class Invoice {

    private int id;
    private int tripId;
    private int reservationId;
    private String payerEmail;
    private String payeeEmail;
    private String type;
    private double amount;
    private String state;

    public Invoice(int tripId, int reservationId, String payerEmail, String payeeEmail, String type, double amount) {
        this.tripId = tripId;
        this.reservationId = reservationId;
        this.payerEmail = payerEmail;
        this.payeeEmail = payeeEmail;
        this.type = type;
        this.amount = amount;
        this.state = "";
    }

    public Invoice(int id, String state) {
        this.id = id;
        this.state = state;
    }

    public int getId() {
        return id;
    }

    public int getTripId() {
        return tripId;
    }

    public String getPayerEmail() {
        return payerEmail;
    }

    public String getPayeeEmail() {
        return payeeEmail;
    }

    public String getType() {
        return type;
    }

    public double getAmount() {
        return amount;
    }

    public String getState() {
        return state;
    }
}

File: Data/Data.Models/Entities/Invoice.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Data.Models.Entities
{
    public class Invoice
    {

        [JsonPropertyName("id")]
        [Key]
        public int Id { get; set; }

        [JsonPropertyName("tripId")]
        public int? TripId { get; set; }

        [JsonPropertyName("reservationId")]
        public int? ReservationId { get; set; }

        [JsonPropertyName("payerEmail")]
        public string PayerEmail { get; set; }

        [JsonPropertyName("payeeEmail")]
        public string PayeeEmail { get; set; }

        [JsonPropertyName("type")]
        public string Type { get; set; }

        [JsonPropertyName("amount")]
        public double Amount { get; set; }

        [JsonPropertyName("state")]
        [Required]
        public string State { get; set; }

    }
}